### PR TITLE
update package files and switch to github actions

### LINKED ISSUE #48: Implement APE17 for repository files
Update to APE17 for the repository files. Update to the astropy package template.
https://docs.astropy.org/projects/package-template/en/latest/ape17.html

This should fix the flake8 error that has recently emerged in the travis testing.  And put this repository on a firmer footing for the future.

### PR DESCRIPTION
**Pull request by karllark**
Update to the package files to the newer APE17 standard.
Migrate from travis ci to github actions
Migrate from coveralls to codecov

Closes #48.
Partially addresses #53.

### CODE REVIEW

**Comment by karllark** [comment]
@mdecleir : I will merge this PR soon.  Tagging you in case you want to look it over at some point to see the changes.  No formal review needed.

File: measure_extinction/merge_obsspec.py [hunk 1]
```diff
@@ -82,7 +86,7 @@ def merge_iue_obsspec(obstables, output_resolution=500):
         cfluxes = ctable["FLUX"].data
         cnpts = ctable["NPTS"].data
         for k in range(n_waves):
-            indxs, = np.where(
+            (indxs,) = np.where(
```
[the remark this answers is not in the document]
Comment (karllark) [reply measure_extinction/merge_obsspec.py:89]: black adds them to make it clear that a tuple is returned and the first element unpacked into indxs.

File: measure_extinction/__init__.py [hunk 2]
```diff
@@ -6,24 +6,4 @@
 from ._astropy_init import *   # noqa
 # ----------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (karllark) [reply measure_extinction/__init__.py:8]: It is importing stuff from _astropy_init.  This happens for all imports in measure_extinction.

[the remark this answers is not in the document]
Comment (karllark) [reply measure_extinction/__init__.py:8]: part of the astropy package template

File: docs/index.rst [hunk 3]
```diff
@@ -49,26 +49,21 @@ trace if it is necessary.
...
-* `Coding Guidelines <http://docs.astropy.io/en/latest/development/codeguide.html>`_
-
-* `Try the development version <http://astropy.readthedocs.io/en/stable/development/workflow/get_devel_version.html>`_
-
-* `Developer Documentation <http://docs.astropy.org/en/latest/#developer-documentation>`_
+* `Coding Guidelines <https://docs.astropy.org/en/stable/development/codeguide.html>`_
 
+* `Developer Documentation <https://docs.astropy.org/en/stable/#developer-documentation>`_
 
 For the complete list of contributors please see the `dust_extinction
```
[the remark this answers is not in the document]
Comment (karllark) [reply docs/index.rst:68]: I will fix that in a separate PR (about to be opened).

File: measure_extinction/conftest.py [hunk 4]
```diff
@@ -1,48 +1,49 @@
...
+        PYTEST_HEADER_MODULES.pop('Pandas', None)
+        PYTEST_HEADER_MODULES['scikit-image'] = 'skimage'
+
+        from . import __version__
+        packagename = os.path.basename(os.path.dirname(__file__))
+        TESTED_VERSIONS[packagename] = __version__
+
+# Uncomment the last two lines in this block to treat all DeprecationWarnings as
+# exceptions. For Astropy v2.0 or later, there are 2 additional keywords,
+# as follow (although default should work for most cases).
```
[the remark this answers is not in the document]
Comment (karllark) [reply measure_extinction/conftest.py:40]: Same separate PR.  :-)

File: measure_extinction/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mdecleir) [reply measure_extinction/__init__.py:8]: > It is importing stuff from _astropy_init. This happens for all imports in measure_extinction.

Do you mean that with this file stuff from _astropy_init is imported every time you import a measure_extinction function/class? Or when exactly will this __init__.py file be run?

**Pull request merged**